judge/views/user.py

```diff
@@ -270,6 +275,88 @@ def get_queryset(self):
         return queryset.order_by('-sticky', '-publish_on').prefetch_related('authors__user')
 
 
+class UserCommentPage(CustomUserMixin, DiggPaginatorMixin, ListView):
+    template_name = 'user/comment.html'
+    model = Comment
+    paginate_by = 10
+    context_object_name = 'comments'
+    title = None
+
+    def get_queryset(self):
+        queryset = (Comment.objects.filter(author=self.user, hidden=False)
+                    .prefetch_related('author__user', 'author__display_badge')
+                    .defer('author__about').order_by('-id').annotate(revisions=Count('versions')))
+
+        user = self.request.user
+
+        if user.is_authenticated:
+            queryset = queryset.annotate(vote_score=Coalesce(RawSQLColumn(CommentVote, 'score'), Value(0)))
+            unique_together_left_join(queryset, CommentVote, 'comment', 'voter', self.request.profile.id)
+
+        problem_cache = CacheDict(lambda code: Problem.objects.defer('description', 'summary').get(code=code))
+        solution_cache = CacheDict(lambda code: Solution.objects.defer('content').get(problem__code=code))
+        contest_cache = CacheDict(lambda key: Contest.objects.defer('description').get(key=key))
+        blog_cache = CacheDict(lambda id: BlogPost.objects.defer('summary', 'content').get(id=id))
+        problemtag_cache = CacheDict(lambda code: TagProblem.objects.get(code=code))
+
+        problem_access = CacheDict(lambda code: problem_cache[code].is_accessible_by(user))
+        solution_access = CacheDict(lambda code: problem_access[code] and solution_cache[code].is_accessible_by(user))
+        contest_access = CacheDict(lambda key: contest_cache[key].is_accessible_by(user))
+        blog_access = CacheDict(lambda id: blog_cache[id].can_see(user))
```

Comment:
this is somewhat familiar with the current get recent comments function, could we create a function to avoid code duplication

Reply:
I'm working on it.